### ISSUE TITLE
Add skeleton content to learning resources page

### ISSUE THREAD

**Issue by drewclem** [drewclem/protege#94]
The learning resources page is making multiple calls to retrieve images for the resources listed. It's not terribly slow, but it could be improved by adding a skeleton loader that mimics the shape of the content that will render later. Example below.

The loading state is being set in two places in Contributors.js. I'm not sure which would be the better place for the loader to go. Surprise me!

Ping me if you have any questions or concerns.

## Definition of done

- [ ] loading hook is set to false on page load
- [ ] a skeleton loader renders on page load and dissolves into content
- [ ] loading hook is set to false after images resolve

![image](https://user-images.githubusercontent.com/36208622/87060626-c034bd80-c1d8-11ea-995e-6dcb2298659e.png)

**Comment by mrsantons** [comment]
Ill take this one! Woop Woop

**Comment by drewclem** [comment]
All yours!

**Comment by pickleat** [comment]
Hey @mrsantons just wanted to check in and see if you need any help here.

**Comment by DZuz14** [comment]
Why not something simple like one line of text that says, "Checking for contributors...", and don't show anything until all contributors are available?

**Comment by DZuz14** [comment]
Something simple could allow more important things to be worked on.

**Comment by drewclem** [comment]
It's partially a personal preference on my part - but we'll also be using skeleton loaders in other places in the future so this is also kind of getting that standard in place.

It gets funky if you throttle the connection down and you end up with some cards rendered and some not - skeleton loader until all are finished would fix that. I like the idea of a timer though!

**Comment by DZuz14** [comment]
Yeah, skeleton seems easier actually. I might look into this is I have some time. I can look at my schedule in a bit.

**Comment by DZuz14** [comment]
From what I saw from `react-placeholder`, it wouldn't be too hard to do something like that ourselves.

Quick off the top of my head idea:

## Simple Skeleton
```
<Skeleton type="text" ready={someCondition} rows={8}>
    ...whatever component(s) here
</Skeleton>
```
- `text | round | rect | video | img | ...etc`

Passing one of these types as a prop would result in a default pre-styled component for that type to be rendered.

## More Complex Skeleton
```
import Skeleton, { Video, Round, Text } from './Skeleton'

<Skeleton ready={someCondition}>
    <div>
      <Video />
   </div>

    <div>
       <Round size={10} />
       <Text rows={4} />
    </div>
</Skeleton>
```

Could be fun to mess around with if keeping external libraries low is priority. However, there are also libraries out there that you might want to look at.

- https://github.com/buildo/react-placeholder
- https://github.com/dvtng/react-loading-skeleton

**Comment by DZuz14** [comment]
@mrsantons @drewclem @pickleat

**Comment by drewclem** [comment]
My initial thought was to just build a skeleton from scratch anytime we needed one - but I like the idea of having a lib we can just drop in to handle it for us.

Would save us a ton of time in the long run with fewer things we have to worry about breaking.

Thanks for this input @DZuz14, I'm liking what I see what react-loading-skeleton

**Comment by pickleat** [comment]
Marie got a new job and isn't able to take this one if someone else wants to jump in!

**Comment by DZuz14** [comment]
If you guys have anything more difficult that needs solving I'd be glad to hear about what it is. I'll have to pass on this one.

**Comment by DZuz14** [comment]
^ not saying this isn't difficult, but it seems better suited towards someone looking for experience.

**Comment by saramccombs** [comment]
I'd be game to work on this. Would help me dive into a React concept I haven't touched before. But, also, totally not wanting to horde the issues from others since I've previously contributed.

**Comment by drewclem** [comment]
@saramccombs It's all yours!

**Comment by saramccombs** [comment]
Awesome, I'm on it. Feel free to assign it to me.

**Comment by saramccombs** [comment]
Update: Working on it. Got a job (!!!!) in the middle and had to press pause for a moment.

**Comment by drewclem** [comment]
@saramccombs Woohoo!! Congratulations!

**Comment by drewclem** [comment]
Congrats on the job!

**Comment by saramccombs** [comment]
Assuming when @nickytonline's PR has merged the need for this feature ceases as he mentioned in #116. Can you confirm this?

While I've been significantly behind on this contribution due to family emergencies and a new job, this issue seems to be rather lucky for those who are job hunting. ❤️

**Comment by drewclem** [comment]
That's probably true. The learning resources could use one though. It's making a call out to retrieve images from multiple sources.

@saramccombs I'll assign you and rename the issue and we'll see who else we can get hired!

**Comment by nickytonline** [comment]
As @drewclem mentioned this would be a good PR for other pages that are loading data dynamically.

**Comment by sethburtonhall** [comment]
@drewclem I will take this one. I plan to integrate react-loading-skeleton and put it to use in "Learning Resources", then we can move into using it in other pages that load data dynamically.

**Comment by sethburtonhall** [comment]
@drewclem I am not seeing where learning resources is making a call to retrieve images, looks to be hard coded.

https://github.com/drewclem/protege/issues/94#issuecomment-682288041

**Comment by sethburtonhall** [comment]
Submitted a PR with a solution. Let me know what you think or if you want anything adjusted.

**Comment by drewclem** [comment]
Closed with #225 Thanks @sethburtonhall
